### ISSUE TITLE
Failure on Raspberry Pi OS build from 20201202

### ISSUE THREAD

**Issue by alcarnes** [openmediavault/openmediavault#879]
### Description of issue/question
When following the raspberry pi installation guide the installation fails with the following error:
```
E: Failed to fetch http://packages.openmediavault.org/public/dists/usul/InRelease  451  Unavailable For Legal Reasons [IP: 195.46.39.1 80]
E: The repository 'http://packages.openmediavault.org/public usul InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
```

### Steps to reproduce issue
(Include debug logs if possible and relevant.)

I used the raspberry pi os lite download available today from this location:
https://www.raspberrypi.org/software/operating-systems/#raspberry-pi-os-32-bit

I am following the installation guide here: 
https://github.com/OpenMediaVault-Plugin-Developers/docs/blob/master/Adden-B-Installing_OMV5_on_an%20R-PI.pdf

After running the following command: 
`wget -O - https://github.com/OpenMediaVault-Plugin-Developers/installScript/raw/master/install | sudo bash`

The output of the script execution is as follows:
```
wget -O - https://github.com/OpenMediaVault-Plugin-Developers/installScript/raw/master/install | sudo bash
[sudo] password for acarnes: --2020-12-11 18:27:33--  https://github.com/OpenMediaVault-Plugin-Developers/installScript/raw/master/install
Resolving github.com (github.com)... 192.30.255.113
Connecting to github.com (github.com)|192.30.255.113|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/OpenMediaVault-Plugin-Developers/installScript/master/install [following]
--2020-12-11 18:27:34--  https://raw.githubusercontent.com/OpenMediaVault-Plugin-Developers/installScript/master/install
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.52.133
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.52.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 16046 (16K) [text/plain]
Saving to: ‘STDOUT’

-                      100%[============================>]  15.67K  --.-KB/s    in 0.002s  

2020-12-11 18:27:34 (6.48 MB/s) - written to stdout [16046/16046]

Current / permissions = 755
New / permissions = 755
Forcing IPv4 only for apt...
Updating repos before installing...
Hit:1 http://archive.raspberrypi.org/debian buster InRelease
Hit:2 http://raspbian.raspberrypi.org/raspbian buster InRelease
Reading package lists... Done
Installing lsb_release...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
0 upgraded, 0 newly installed, 1 reinstalled, 0 to remove and 0 not upgraded.
Need to get 27.8 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://mirror.web-ster.com/raspbian/raspbian buster/main armhf lsb-release all 10.2019051400+rpi1 [27.8 kB]
Fetched 27.8 kB in 1s (46.2 kB/s)
(Reading database ... 40568 files and directories currently installed.)
Preparing to unpack .../lsb-release_10.2019051400+rpi1_all.deb ...
Unpacking lsb-release (10.2019051400+rpi1) over (10.2019051400+rpi1) ...
Setting up lsb-release (10.2019051400+rpi1) ...
Processing triggers for man-db (2.8.5-2) ...
usul :: 5
Adding Debian signing keys...
Executing: /tmp/apt-key-gpghome.glJBGrprD2/gpg.1.sh --no-tty --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys AA8E81B4331F7F50
gpg: key EDA0D2388AE22BA9: public key "Debian Security Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Executing: /tmp/apt-key-gpghome.TEOWDR3Adm/gpg.1.sh --no-tty --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 112695A0E562B32A
gpg: key 4DFAB270CAA96DFA: public key "Debian Security Archive Automatic Signing Key (10/buster) <ftpmaster@debian.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Executing: /tmp/apt-key-gpghome.rfBfkcexk5/gpg.1.sh --no-tty --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 04EE7237B7D453EC
gpg: key E0B11894F66AEC98: public key "Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Executing: /tmp/apt-key-gpghome.5DeRLdfcIe/gpg.1.sh --no-tty --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 648ACFD622F3D138
gpg: key DC30D7C23CBBABEE: public key "Debian Archive Automatic Signing Key (10/buster) <ftpmaster@debian.org>" imported
gpg: Total number processed: 1
gpg:               imported: 1
Installing monit from raspberrypi repo...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Suggested packages:
  default-mta | postfix | mail-transport-agent sysvinit-core
The following NEW packages will be installed:
  monit
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 291 kB of archives.
After this operation, 847 kB of additional disk space will be used.
Get:1 http://mirror.web-ster.com/raspbian/raspbian buster/main armhf monit armhf 1:5.25.2-3 [291 kB]
Fetched 291 kB in 1s (486 kB/s)
Selecting previously unselected package monit.
(Reading database ... 40568 files and directories currently installed.)
Preparing to unpack .../monit_1%3a5.25.2-3_armhf.deb ...
Unpacking monit (1:5.25.2-3) ...
Setting up monit (1:5.25.2-3) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for systemd (241-7~deb10u5+rpi1) ...
Install prerequisites...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
dirmngr is already the newest version (2.2.12-1+rpi1+deb10u1).
gnupg is already the newest version (2.2.12-1+rpi1+deb10u1).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Installing openmediavault required packages...
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  ssl-cert
Suggested packages:
  procmail postfix-mysql postfix-pgsql postfix-ldap postfix-pcre postfix-lmdb
  postfix-sqlite sasl2-bin | dovecot-common libsasl2-modules | dovecot-common postfix-cdb
  mail-reader ufw postfix-doc openssl-blacklist
The following NEW packages will be installed:
  postfix ssl-cert
0 upgraded, 2 newly installed, 0 to remove and 0 not upgraded.
Need to get 1423 kB of archives.
After this operation, 3750 kB of additional disk space will be used.
Get:1 http://mirror.web-ster.com/raspbian/raspbian buster/main armhf ssl-cert all 1.0.39 [20.8 kB]
Get:2 http://mirror.web-ster.com/raspbian/raspbian buster/main armhf postfix armhf 3.4.14-0+deb10u1 [1402 kB]
Fetched 1423 kB in 1s (1179 kB/s) 
Preconfiguring packages ...
Selecting previously unselected package ssl-cert.
(Reading database ... 40613 files and directories currently installed.)
Preparing to unpack .../ssl-cert_1.0.39_all.deb ...
Unpacking ssl-cert (1.0.39) ...
Selecting previously unselected package postfix.
Preparing to unpack .../postfix_3.4.14-0+deb10u1_armhf.deb ...
Unpacking postfix (3.4.14-0+deb10u1) ...
Setting up ssl-cert (1.0.39) ...
Setting up postfix (3.4.14-0+deb10u1) ...
Adding group `postfix' (GID 115) ...
Done.
Adding system user `postfix' (UID 109) ...
Adding new user `postfix' (UID 109) with group `postfix' ...
Not creating home directory `/var/spool/postfix'.
Creating /etc/postfix/dynamicmaps.cf
Adding group `postdrop' (GID 116) ...
Done.
setting myhostname: piomv.localdomain
setting alias maps
setting alias database
mailname is not a fully qualified domain name.  Not changing /etc/mailname.
setting destinations: $myhostname, piomv, localhost.localdomain, , localhost
setting relayhost: 
setting mynetworks: 127.0.0.0/8 [::ffff:127.0.0.0]/104 [::1]/128
setting mailbox_size_limit: 0
setting recipient_delimiter: +
setting inet_interfaces: all
setting inet_protocols: all
/etc/aliases does not exist, creating it.
WARNING: /etc/aliases exists, but does not have a root alias.

Postfix (main.cf) is now set up with a default configuration.  If you need to 
make changes, edit /etc/postfix/main.cf (and others) as needed.  To view 
Postfix configuration values, see postconf(1).

After modifying main.cf, be sure to run 'service postfix reload'.

Running newaliases
Created symlink /etc/systemd/system/multi-user.target.wants/postfix.service → /lib/systemd/system/postfix.service.
Processing triggers for systemd (241-7~deb10u5+rpi1) ...
Processing triggers for man-db (2.8.5-2) ...
Processing triggers for rsyslog (8.1901.0-1) ...
Adding openmediavault repo and key...
--2020-12-11 18:29:33--  http://packages.openmediavault.org/public/archive.key
Resolving packages.openmediavault.org (packages.openmediavault.org)... 195.46.39.1
Connecting to packages.openmediavault.org (packages.openmediavault.org)|195.46.39.1|:80... connected.
HTTP request sent, awaiting response... 451 Unavailable For Legal Reasons
2020-12-11 18:29:33 ERROR 451: Unavailable For Legal Reasons.

gpg: no valid OpenPGP data found.
Updating repos...
Err:1 http://packages.openmediavault.org/public usul InRelease
  451  Unavailable For Legal Reasons [IP: 195.46.39.1 80]
Hit:2 http://archive.raspberrypi.org/debian buster InRelease                               
Hit:3 http://raspbian.raspberrypi.org/raspbian buster InRelease                            
Reading package lists... Done
E: Failed to fetch http://packages.openmediavault.org/public/dists/usul/InRelease  451  Unavailable For Legal Reasons [IP: 195.46.39.1 80]
E: The repository 'http://packages.openmediavault.org/public usul InRelease' is not signed.
N: Updating from such a repository can't be done securely, and is therefore disabled by default.
N: See apt-secure(8) manpage for repository creation and user configuration details.
failed to update apt repos.
```
Can you please provide some insight as to what I might be doing wrong? I performing this on a raspberry pi 4 8gb

**Comment by alcarnes** [comment]
I tried the script execution against Raspbian OS from 20200213 and got the same result.  However, I did do an update/upgrade before running the script.

**Comment by votdev** [comment]
```
In computer networking, HTTP 451 Unavailable For Legal Reasons is an error status code of the HTTP protocol to be displayed when the user requests a resource which cannot be served for legal reasons, such as a web page censored by a government.
```
Where are you from?

**Comment by alcarnes** [comment]
I'm in Seattle, Washington. I am not using a VPN service and I set my location information in the raspberry pi prior to attempting to install through raspi-config.  Thank you for your response.

**Comment by votdev** [comment]
> I'm in Seattle, Washington. I am not using a VPN service and I set my location information in the raspberry pi prior to attempting to install through raspi-config. Thank you for your response.

Does it still not work? I'm wondering because this would mean that the IP or FQDN of OMV is blocked in the States or at least by your provider. Can not reproduce this here.

**Comment by ryecoaaron** [comment]
No problems in Wisconsin.

**Comment by alcarnes** [comment]
I haven't had the chance to retry this but I am connecting through Comcast in Washington.  I will give it a try tomorrow. Thanks for your replies.

**Comment by alcarnes** [comment]
I connected via my phone as a hotspot and it installed correctly.   As votdev said, the IP or FQDN of OMV seems to be blocked by Comcast at least in the Seattle Washington area.  Does anybody have suggestions on how to get this rectified with them? I could install a VPN on the system and perhaps enable that when needing to update if I have to.

**Comment by votdev** [comment]
It's really difficult or not possible to contact Comcast. I tried their Twitter account ComcastCares, but i really don't expect any response or action.

**Comment by votdev** [comment]
@alcarnes I've contacted ComcastCares via Twitter and the told me the following:
```
It is possible that your customer is subscribed to our XFi security package and it may be blocking the domain for him specifically. They should be easily resolved by your customer contacting our Customer Security Assurance Team directly at 1-888-565-4329. They are open from 6:00 AM - 2:00 AM ET/Seven days a week.
```

**Comment by alcarnes** [comment]
After some investigation, I found out that I am using a dns provider that does some ip filtering.  For some reason https://packages.openmediavault.org/public was marked as a source for phishing attacks and therefore was being blacklisted.  I manually added it to my whitelist but it was still being blocked for some reason.  I contacted the DNS provider (safedns.com)  and they reviewed the site and the fact that the whitelist entry was still not successful.  They made some server changes and things started working again.

Thanks everybody for the information.  I am going to close this as resolved.
